tests/components/aemet/test_config_flow.py

```diff
@@ -0,0 +1,70 @@
+"""Define tests for the AEMET OpenData config flow."""
+
+from unittest.mock import patch
+
+import requests_mock
+
+from homeassistant import data_entry_flow
+from homeassistant.components.aemet.const import DOMAIN
+from homeassistant.config_entries import ENTRY_STATE_LOADED, SOURCE_USER
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+import homeassistant.util.dt as dt_util
+
+from .util import aemet_requests_mock
+
+CONFIG = {
+    CONF_NAME: "aemet",
+    CONF_API_KEY: "foo",
+    CONF_LATITUDE: 40.30403754,
+    CONF_LONGITUDE: -3.72935236,
+}
+
+
+async def test_form(hass):
+    """Test that the form is served with valid input."""
+
+    now = dt_util.parse_datetime("2021-01-09 12:00:00+00:00")
+    with patch("homeassistant.util.dt.now", return_value=now), patch(
+        "homeassistant.util.dt.utcnow", return_value=now
+    ), requests_mock.mock() as _m:
+        aemet_requests_mock(_m)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == SOURCE_USER
+        assert result["errors"] == {}
+
+        result = await hass.config_entries.flow.async_init(
```

Comment:
We normally patch the integration set up function on all create entry results to avoid setting up the whole integration. See the scaffold script tests template for example.

Reply:
I don't believe this is resolved, I think Martin is asking for the setup and setup_entry functions to be patched, see https://github.com/home-assistant/core/blob/061d9c5293d1e049ee3d407aafa718891822f2dc/script/scaffold/templates/config_flow/tests/test_config_flow.py#L21_L26

Reply:
@MatthewFlamm yeah, I wasn't sure about what he wanted...

Reply:
> We normally patch the integration set up function on all create entry results to avoid setting up the whole integration. See the scaffold script tests template for example.

@MartinHjelmare Is this what you wanted? https://github.com/home-assistant/core/pull/45074/commits/65861c12a20c45939d97f96ad0c0caac0b87166c